keras_cv/visualization/plot_image_gallery.py

```diff
@@ -92,27 +89,87 @@ def unpackage_tfds_inputs(inputs):
             "to be true."
         )
 
-    fig = plt.figure(figsize=(cols * scale, rows * scale))
-    fig.tight_layout()  # Or equivalently,  "plt.tight_layout()"
-    plt.subplots_adjust(wspace=0, hspace=0)
-    plt.margins(x=0, y=0)
-    plt.axis("off")
+    def unpack_images(inputs):
+        return inputs["image"]
+
+    # Calculate appropriate number of rows and columns
+    if rows is None and cols is None:
+        if isinstance(images, tf.data.Dataset):
+            sample = next(iter(images.take(1)))
+            sample_shape = sample["image"].shape
+            if len(sample_shape) == 4:
+                batch_size = sample_shape[0]
+            else:
+                raise ValueError(
+                    "Passed `tf.data.Dataset` does not appear to be batched. Please batch using the `.batch().`"
+                )
+
+            images = images.map(unpack_images)
+            images = images.take(batch_size)
+            images = next(iter(images))
+        else:
+            sample_shape = images.shape
+            if len(sample_shape) == 4:
+                batch_size = sample_shape[0]
+            else:
+                raise ValueError(
+                    f"Passed '`{type(images)}`' does not appear to be batched. Please batch using the `.batch()."
+                )
+
+    elif rows is not None and cols is not None:
+        if isinstance(images, tf.data.Dataset):
+            batch_size = rows * cols
+
+            sample = next(iter(images.take(1)))
+            sample_shape = sample["image"].shape
+
+            if len(sample_shape) == 4:
+                images = images.unbatch()
+
+            images = images.ragged_batch(batch_size=batch_size)
+
+            images = images.map(unpack_images)
+            images = images.take(batch_size)
+            images = next(iter(images))
+        else:
+            batch_size = rows * cols
+            images = images[:batch_size, ...]
+    else:
+        raise ValueError(
```

Comment:
This error seems like it will be thrown iff one of `rows` or `cols` is set, but not both. Let's update the error message to reflect that.

Reply:
Done!